src/Codeception/Module/Db.php

```diff
@@ -808,7 +808,9 @@ private function addInsertedRow(string $table, array $row, $id): void
     {
         $primaryKey = $this->_getDriver()->getPrimaryKey($table);
         $primary = [];
-        if ($primaryKey !== []) {
+        if (count($primaryKey) == 1 && is_null($row[$primaryKey[0]])) {
```

Comment:
Why not check for "all pk values are null" instead of checking the size is 1 and the single value is null ?